src/core/qgsvectorlayer.cpp

```diff
@@ -4335,6 +4341,27 @@ QString QgsVectorLayer::getStyleFromDatabase( const QString& styleId, QString &m
   return getStyleByIdMethod( mDataSource, styleId, msgError );
 }
 
+bool QgsVectorLayer::deleteStyleFromDatabase( const QString& styleId, QString &msgError )
+{
+  QgsProviderRegistry * pReg = QgsProviderRegistry::instance();
+  QLibrary *myLib = pReg->providerLibrary( mProviderKey );
```

Comment:
Thanks @nyalldawson.
The pattern I was using (without the `QScopedPointer`) is still used on many places. I've updated all usages of the old pattern just in `src/core/qgsvectorlayer.cpp` to this more safer alternative, using `QScopedPointer` without an explicit delete of the pointer.